spec/jobs/import_url_job_spec.rb

```diff
@@ -18,50 +18,41 @@
     double('response').tap do |http_res|
       allow(http_res).to receive(:start).and_yield
       allow(http_res).to receive(:content_type).and_return('image/png')
-      allow(http_res).to receive(:read_body).and_yield(File.open(File.expand_path('../../fixtures/world.png', __FILE__)).read)
+      allow(http_res).to receive(:read_body).and_yield(File.open(File.expand_path(file_path, __FILE__)).read)
     end
   end
 
   subject(:job) { ImportUrlJob.new(generic_file.id) }
 
   it "should have no content at the outset" do
-    expect(generic_file.content.size).to be_nil
+    expect(generic_file.original_file).to be_nil
   end
 
-  ### TODO - Needs Refactoring after Sufia is Broken up
-  # context "after running the job" do
-  #   before do
-  #     s1 = double('content deposit event')
-  #     allow(ContentDepositEventJob).to receive(:new).with(generic_file.id, 'jilluser@example.com').and_return(s1).once
-  #     expect(CurationConcerns.queue).to receive(:push).with(s1).once
-  #
-  #     s2 = double('characterize')
-  #     allow(CharacterizeJob).to receive(:new).with(generic_file.id).and_return(s2)
-  #     expect(CurationConcerns.queue).to receive(:push).with(s2).once
-  #
-  #     expect(CurationConcerns::GenericFileActor).to receive(:virus_check).and_return(false)
-  #   end
-  #
-  #   xit "should create a content datastream" do
-  #     expect_any_instance_of(Net::HTTP).to receive(:request_get).with(file_hash).and_yield(mock_response)
-  #     job.run
-  #     expect(generic_file.reload.content.size).to eq 4218
-  #     expect(user.mailbox.inbox.first.last_message.body).to eq("The file (#{file_path}) was successfully imported.")
-  #   end
-  # end
-  #
-  # context "when the file has a virus" do
-  #   before do
-  #     s1 = double('content deposit event')
-  #     allow(ContentDepositEventJob).to receive(:new).with(generic_file.id, 'jilluser@example.com').never
-  #
-  #     s2 = double('characterize')
-  #     allow(CharacterizeJob).to receive(:new).with(generic_file.id).never
-  #   end
-  #   xit "should abort if virus check fails" do
-  #     allow(CurationConcerns::GenericFileActor).to receive(:virus_check).and_raise(Sufia::VirusFoundError.new('A virus was found'))
-  #     job.run
-  #     expect(user.mailbox.inbox.first.subject).to eq("File Import Error")
-  #   end
-  # end
+  context "after running the job" do
+    before do  
+      s1 = double('characterize')
+      allow(CharacterizeJob).to receive(:new).with(generic_file.id).and_return(s1)
+      expect(CurationConcerns.queue).to receive(:push).with(s1).once
+      expect(CurationConcerns::VirusDetectionService).to receive(:run).and_return(false)
+    end
+  
+    it "should create a content datastream" do
+      expect_any_instance_of(Net::HTTP).to receive(:request_get).with(file_hash).and_yield(mock_response)
+      job.run
+      expect(generic_file.reload.original_file.size).to eq 4218
+    end
+  end
+  
+  context "when the file has a virus" do
```

Comment:
Unless there is code in this Job that does something distinctive with virus
checking, you should remove it.

On Fri, Jul 10, 2015 at 10:28 AM, Eliot Jordan <notifications@github.com>
wrote:

> In spec/jobs/import_url_job_spec.rb
> <https://github.com/projecthydra-labs/curation_concerns/pull/71#discussion_r34376348>
> :
>
> > +  context "after running the job" do
> > +    before do
> > +      s1 = double('characterize')
> > +      allow(CharacterizeJob).to receive(:new).with(generic_file.id).and_return(s1)
> > +      expect(CurationConcerns.queue).to receive(:push).with(s1).once
> > +      expect(CurationConcerns::VirusDetectionService).to receive(:run).and_return(false)
> > +    end
> > +
> > +    it "should create a content datastream" do
> > +      expect_any_instance_of(Net::HTTP).to receive(:request_get).with(file_hash).and_yield(mock_response)
> > +      job.run
> > +      expect(generic_file.reload.original_file.size).to eq 4218
> > +    end
> > +  end
> > +
> > +  context "when the file has a virus" do
>
> I think you're right. Is this even necessary?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projecthydra-labs/curation_concerns/pull/71/files#r34376348>
> .
>